pkg/asset/manifests/storageclassconfig.go

```diff
@@ -0,0 +1,135 @@
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	azuretypes "github.com/openshift/installer/pkg/types/azure"
+)
+
+var (
+	masterStorageClassConfig = filepath.Join(manifestDir, "storageclass-01-master.yaml")
+	workerStorageClassConfig = filepath.Join(manifestDir, "storageclass-02-worker.yaml")
+)
+
+// StorageClass generates the storageclass yaml files.
+type StorageClass struct {
+	FileList []*asset.File
+}
+
+var _ asset.WritableAsset = (*StorageClass)(nil)
+
+// Name returns a human friendly name for the asset.
+func (*StorageClass) Name() string {
+	return "Storage Class"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate the
+// asset.
+func (*StorageClass) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+		&installconfig.ClusterID{},
+		// PlatformCredsCheck just checks the creds (and asks, if needed)
+		// We do not actually use it in this asset directly, hence
+		// it is put in the dependencies but not fetched in Generate
+		&installconfig.PlatformCredsCheck{},
+	}
+}
+
+// Generate generates the storage class config file.
+func (sc *StorageClass) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	clusterID := &installconfig.ClusterID{}
+	dependencies.Get(installConfig, clusterID)
+
+	// It's possible to have different storage class types for both the
+	// master pool and the worker pool. For the master pool it is simple to
+	// determine the storage class. For the worker pool, this is a bit of a
+	// hacky solution. As soon as a storage class is determined, we stop
+	// and assume that is the storage class. This should probably be added
+	// as a validation.
+	switch installConfig.Config.Platform.Name() {
+	case azuretypes.Name:
+		masterPool := installConfig.Config.ControlPlane.Platform.Azure
+		if masterPool.OSDisk.DiskEncryptionSet != nil {
+			storageClass := azureStorageClass(masterPool, "master")
+			configData, err := yaml.Marshal(storageClass)
+			if err != nil {
+				return errors.Wrapf(err, "failed to encode %s master manifest", sc.Name())
+			}
+			sc.FileList = append(sc.FileList, &asset.File{
+				Filename: masterStorageClassConfig,
+				Data:     configData,
+			})
+		}
+
+		workerPool := installConfig.Config.Compute
+		for _, pool := range workerPool {
+			wpool := pool.Platform.Azure
+			if wpool.OSDisk.DiskEncryptionSet != nil {
+				storageClass := azureStorageClass(wpool, "worker")
+				configData, err := yaml.Marshal(storageClass)
+				if err != nil {
+					return errors.Wrapf(err, "failed to encode %s worker manifest", sc.Name())
+				}
+				sc.FileList = append(sc.FileList, &asset.File{
+					Filename: workerStorageClassConfig,
+					Data:     configData,
+				})
+				break
+			}
+		}
+	}
+
+	return nil
+}
+
+func azureStorageClass(pool *azuretypes.MachinePool, name string) *storagev1.StorageClass {
```

Comment:
We're in process of making the CSI storage class as default in 4.11, hence this should generate a CSI storage class ([like this one](https://github.com/openshift/azure-disk-csi-driver-operator/blob/master/assets/storageclass.yaml) + encryption params).

https://github.com/openshift/azure-disk-csi-driver-operator/pull/47

Reply:
Updated.